addons/test_website/tests/test_controller_args.py

```diff
@@ -29,3 +31,17 @@ def test_crawl_args(self):
         req = self.url_open('/ignore_args/kw?a=valueA&b=valueB')
         self.assertEqual(req.status_code, 200)
         self.assertEqual(req.json(), {'a': 'valueA', 'kw': {'b': 'valueB'}})
+
+    def test_route_type(self):
+        response = self.url_open('/test_route_json_http?blabla=qsd')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.text, "{'blabla': 'qsd'}")
+
+        response = self.url_open('/test_route_json_http', data=b"blabla=qsd", headers={'Content-type': 'application/x-www-form-urlencoded'})
```

Comment:
For new code, I like to keep lines under 100-120 chars, it's less annoying to view on github split mode (no wrapping), and nicer on the eyes of readers. Thanks! ;-)